### PR TITLE
FABB-69: Return `HTTP 404` when associate a non-existent Figma File or Node

### PR DESCRIPTION
**Pull request by akostevich-atlassian**
## Changes

- **feat:** Updates the `POST entities/associateEntity` endpoint to return `HTTP 404 Not Found` when a Figma File or Node does not exist.
- **refactor:** Applies minor improvements to `FigmaService`.

### CODE REVIEW

File: src/usecases/associate-design-use-case.ts [hunk 1]
```diff
@@ -46,6 +49,8 @@ export const associateDesignUseCase = {
 				jiraService.getIssue(associateWith.id, connectInstallation),
 			]);
 
+			if (!design) throw new FigmaDesignNotFoundUseCaseResultError();
```
Comment (akostevich-atlassian) [inline src/usecases/associate-design-use-case.ts:52]: **note:** The key change :)

File: src/infrastructure/figma/figma-service.ts [hunk 2]
```diff
@@ -51,15 +49,15 @@ export class FigmaService {
 	};
 
 	/**
-	 * Return Atlassian design for the Figma design with the given ID.
+	 * Returns Atlassian design for the Figma design with the given ID if it exists; otherwise -- `null`.
 	 *
 	 * @throws {DesignNodeNotFoundError} Design node referenced by the designId was not found.
```
Comment (elainefigma) [inline src/infrastructure/figma/figma-service.ts:54]: minor: Does this method still throw a DesignNodeNotFoundError? If no, should we delete this comment?

File: src/infrastructure/figma/figma-service.ts [hunk 3]
```diff
@@ -344,50 +342,91 @@ export class FigmaService {
 		});
 
 	/**
-	 * Return Atlassian design representation for the given Figma File.
-	 *
-	 * @throws {UnauthorizedFigmaServiceError} Not authorized to access Figma.
+	 * Returns Atlassian design representation for the given Figma File if it exists; otherwise -- `null`.
```
Comment (elainefigma) [inline src/infrastructure/figma/figma-service.ts:345]: minor: Can this method still throw an UnauthorizedFigmaServiceError i.e. if the user is not logged in to Figma? I have a similar question for the getDesignForNode method as well.

**Review by elainefigma — APPROVED** [review]
Thank you for fixing this! This will also help clean up our error monitors for these endpoints.

File: src/infrastructure/figma/figma-service.ts (hunk 2 above)
Comment (akostevich-atlassian) [reply src/infrastructure/figma/figma-service.ts:54]: Oh, great catch, @elainefigma . Fixed.

File: src/infrastructure/figma/figma-service.ts (hunk 3 above)
Comment (akostevich-atlassian) [reply src/infrastructure/figma/figma-service.ts:344]: These private methods currently do not do error translation but now I am thinking that they should :) I've updated these methods and the code comments accordingly. Thank you, @elainefigma , for the careful review!

**Pull request merged**